File: Database/Assembly_SRPG_JP/ShopParam.cs
// Decompiled with JetBrains decompiler
// Type: SRPG.ShopParam
// Assembly: Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 85BFDF7F-5712-4D45-9CD6-3465C703DFDF
// Assembly location: S:\Desktop\Assembly-CSharp.dll

namespace SRPG
{
  public class ShopParam
  {
    public string iname;
    public ESaleType UpdateCostType;
    public int[] UpdateCosts;

    public bool Deserialize(JSON_ShopParam json)
    {
      if (json == null)
        return false;
      this.iname = json.iname;
      this.UpdateCostType = (ESaleType) json.upd_type;
      this.UpdateCosts = (int[]) null;
      if (json.upd_costs != null && json.upd_costs.Length > 0)
      {
        this.UpdateCosts = new int[json.upd_costs.Length];
        for (int index = 0; index < json.upd_costs.Length; ++index)
          this.UpdateCosts[index] = json.upd_costs[index];
      }
      return true;
    }
  }
}

File: Database/genConvert/ShopParam.py
def ShopParam(json):
    this={}#ShopParamjson)
    #if(json==null)
    #returnfalse
    if 'iname' in json:
        this['iname'] = json['iname']
    if 'upd_type' in json:
        this['UpdateCostType'] = ENUM['ESaleType'][json['upd_type']]
    #this.UpdateCosts=(int)null
    #if(json.upd_costs!=null&&json.upd_costs.Length>0)
        #for(intindex=0index<json.upd_costs.Length++index)
        if 'upd_costs' in json:
            this['UpdateCosts'] = json['upd_costs']
    #returntrue
return this
